kernel/src/table_changes/log_replay/tests.rs

```diff
@@ -0,0 +1,514 @@
+                ..Default::default()
+            }
+            .into(),
+        ])
+        .await;
+
+    let mut commits = get_segment(engine.as_ref(), mock_table.table_root(), 0, None)
+        .unwrap()
+        .into_iter();
+
+    let scanner = LogReplayScanner::prepare_table_changes(
+        commits.next().unwrap(),
+        engine.as_ref(),
+        &get_schema().into(),
+    )
+    .unwrap();
+    assert!(!scanner.has_cdc_action);
+    assert_eq!(scanner.remove_dvs, HashMap::new());
+
+    assert_eq!(
+        result_to_sv(scanner.into_scan_batches(engine, None).unwrap()),
+        &[false; 6]
+    );
+}
+
+#[tokio::test]
+async fn cdc_selection() {
+    let engine = Arc::new(SyncEngine::new());
+    let mut mock_table = LocalMockTable::new();
+    mock_table
+        .commit([
+            Add {
+                path: "fake_path_1".into(),
+                data_change: true,
+                ..Default::default()
+            }
+            .into(),
+            Remove {
+                path: "fake_path_2".into(),
+                data_change: true,
+                ..Default::default()
+            }
+            .into(),
+            Cdc {
+                path: "fake_path_3".into(),
+                data_change: true,
+                ..Default::default()
+            }
+            .into(),
+        ])
+        .await;
+
+    let mut commits = get_segment(engine.as_ref(), mock_table.table_root(), 0, None)
+        .unwrap()
+        .into_iter();
+
+    let scanner = LogReplayScanner::prepare_table_changes(
+        commits.next().unwrap(),
+        engine.as_ref(),
+        &get_schema().into(),
+    )
+    .unwrap();
+    assert!(scanner.has_cdc_action);
+    assert_eq!(scanner.remove_dvs, HashMap::new());
+
+    assert_eq!(
+        result_to_sv(scanner.into_scan_batches(engine, None).unwrap()),
+        &[false, false, true]
+    );
+}
+
+#[tokio::test]
+async fn dv() {
+    let engine = Arc::new(SyncEngine::new());
+    let mut mock_table = LocalMockTable::new();
+
+    let deletion_vector1 = DeletionVectorDescriptor {
+        storage_type: "u".to_string(),
+        path_or_inline_dv: "vBn[lx{q8@P<9BNH/isA".to_string(),
+        offset: Some(1),
+        size_in_bytes: 36,
+        cardinality: 2,
+    };
+    let deletion_vector2 = DeletionVectorDescriptor {
+        storage_type: "u".to_string(),
+        path_or_inline_dv: "U5OWRz5k%CFT.Td}yCPW".to_string(),
+        offset: Some(1),
+        size_in_bytes: 38,
+        cardinality: 3,
+    };
+    mock_table
+        .commit([
+            Add {
+                path: "fake_path_1".into(),
+                data_change: true,
+                ..Default::default()
+            }
+            .into(),
+            Remove {
+                path: "fake_path_1".into(),
+                data_change: true,
+                deletion_vector: Some(deletion_vector1.clone()),
+                ..Default::default()
+            }
+            .into(),
```

Comment:
will we ever remove a (path, dv) and add just a (path, None)? is that valid?

Reply:
Yes, that's a restore operation.

Reply:
I'll document that.

Reply:
A DV purge would also eliminate the DV, but not for the same path which I guess was the real question?

Reply:
Yep I think the question is asking about having the same path